### PR TITLE
ISPN-10149 - Enable checkstyle plugin during the verify maven phase

### PR DESCRIPTION
**Pull request by diegolovison**
I am expecting an error in `server/integration/testsuite/src/test/resources/application-users.properties`

### CODE REVIEW

**Comment by ryanemerson** [comment]
Thanks @diegolovison! One minor thing, please try to remember to add the jira url to the PR description as it makes the reviewers life much easier :smile:

**Comment by ryanemerson** [comment]
@diegolovison It would be good to have this on the `9.4.x` branch, can you issue a backport PR please

**Comment by danberindei** [comment]
@diegolovison I'm reading the PR/JIRA subject as "checkstyle didn't run at all before, this PR is enabling it for the first time", but I'm pretty sure I got checkstyle errors before. Can you explain in more detail what you are changing?

File: build-configuration/pom.xml [hunk 1]
```diff
@@ -260,6 +259,9 @@
       <version.org.wildfly.build-tools>1.2.10.Final</version.org.wildfly.build-tools>
       <version.owasp-dependency-check-plugin>4.0.2</version.owasp-dependency-check-plugin>
 
+      <!-- versionx -->
+      <versionx.com.puppycrawl.tools.checkstyle>8.18</versionx.com.puppycrawl.tools.checkstyle>
```
Comment (danberindei) [inline build-configuration/pom.xml]: Why treat the checkstyle plugin and checkstyle itself differently? They are both needed to build Infinispan, and not needed to run it.

**Comment by diegolovison** [comment]
@danberindei @ryanemerson 
Before my fix the state of ISPN was:
1) Diff versions for `<groupId>com.puppycrawl.tools</groupId>`. In other to fix it, I moved `versionx.com.puppycrawl.tools.checkstyle` to `build-configuration/pom.xml`. Now, `pom.xml` and `checkstyle/pom.xml` can see the same version. `<version.checkstyle>8.1</version.checkstyle>` was removed because it was duplicated and it was an old version.

2) Configure checkstyle plugin
The maven tree structure for the root ISPN pom.xml is:
pom.xml -> infinispan-build-configuration-parent\pom.xml -> 

The maven tree structure for checkstyle project is:
pom.xml -> build-configuration/bom/pom.xml

This is why we saw the same configuration `<artifactId>maven-checkstyle-plugin</artifactId>`plugin in both pom.xml

I will wait for the @danberindei feedback before proceeding with the backport

**Comment by tristantarrant** [comment]
@diegolovison @danberindei can we have a backport ?

**Comment by diegolovison** [comment]
Done. https://github.com/infinispan/infinispan/pull/6982

**Pull request merged**